src/main/java/com/bandwidth/iris/sdk/IrisClient.java

```diff
@@ -1,186 +0,0 @@
-        this.version = version;
-
-        httpClient = new DefaultHttpClient();
-        Credentials credentials = new UsernamePasswordCredentials(userName, password);
-        httpClient.getCredentialsProvider().setCredentials(AuthScope.ANY, credentials);
-        this.baseUrl = "/" + version + "/";
-        this.baseUserUrl = this.baseUrl + "accounts/" + this.accountId + "/";
-
-    }
-
-    public IrisClient(String accountId, String userName, String password) {
-        this(defaultUri, accountId, userName, password, defaultVersion);
-    }
-
-    private <T> T processResponse(String responseBody, Class<T> returnType) throws Exception {
-        T parsedResponse = XmlUtils.fromXml(responseBody, returnType);
-        if (parsedResponse instanceof BaseResponse){
-            BaseResponse baseResponse = (BaseResponse) parsedResponse;
-            if(baseResponse.getResponseStatus() != null){
-                throw new IrisClientException(baseResponse.getResponseStatus().getErrorCode(),
-                        baseResponse.getResponseStatus().getDescription());
-            }
-        }
-        return parsedResponse;
-    }
-
-    public <T> T get(String uri, Class<T> returnType) throws Exception {
-        IrisResponse response = get(uri);
-        return processResponse(response.getResponseBody(), returnType);
-    }
-
-    public IrisResponse get(String uri) throws Exception {
-        HttpGet get = new HttpGet(uri);
-        return executeRequest(get);
-    }
-
-    public <T> T post(String uri, BaseModel data, Class<T> returnType) throws Exception {
-        IrisResponse response = post(uri, data);
-        return processResponse(response.getResponseBody(), returnType);
-    }
-    public IrisResponse post(String uri, BaseModel data) throws Exception {
-        HttpPost post = new HttpPost(uri);
-        StringEntity postBody = new StringEntity(XmlUtils.toXml(data));
-        post.addHeader("Content-Type", "application/xml");
-        post.setEntity(postBody);
-        return executeRequest(post);
-    }
-
-    public IrisResponse delete(String uri) throws Exception {
-        HttpDelete delete = new HttpDelete(uri);
-        return executeRequest(delete);
-    }
-
-    public <T> T put(String uri, BaseModel data, Class<T> returnType) throws Exception {
-        IrisResponse response = put(uri, data);
-        return processResponse(response.getResponseBody(), returnType);
-    }
-    public IrisResponse put(String uri, BaseModel data) throws Exception {
-        HttpPut put = new HttpPut(uri);
-        return executeRequest(put);
-    }
-
-    public void postFile(String uri, File file, String contentType) throws Exception {
-        HttpPost post = new HttpPost(uri);
-        post.setEntity(contentType == null ? new FileEntity(file) : new FileEntity(file, ContentType.parse(contentType)));
-        executeRequest(post);
-    }
-
-    public void putFile(String uri, File file, String contentType) throws Exception {
```

Comment:
the pattern of building the URI came from the other Java SDK (for catapult)

On Tue, Jul 7, 2015 at 1:21 PM, Frank Schmager <notifications@github.com>
wrote:

> In src/main/java/com/bandwidth/iris/sdk/IrisClient.java
> <https://github.com/bandwidthcom/java-bandwidth-iris/pull/1#discussion_r34065524>
> :
>
> > -    public <T> T put(String uri, BaseModel data, Class<T> returnType) throws Exception {
> > -        IrisResponse response = put(uri, data);
> > -        return processResponse(response.getResponseBody(), returnType);
> > -    }
> > -    public IrisResponse put(String uri, BaseModel data) throws Exception {
> > -        HttpPut put = new HttpPut(uri);
> > -        return executeRequest(put);
> > -    }
> > -
> > -    public void postFile(String uri, File file, String contentType) throws Exception {
> > -        HttpPost post = new HttpPost(uri);
> > -        post.setEntity(contentType == null ? new FileEntity(file) : new FileEntity(file, ContentType.parse(contentType)));
> > -        executeRequest(post);
> > -    }
> > -
> > -    public void putFile(String uri, File file, String contentType) throws Exception {
>
> Something is slightly fishy here. When an IrisClient gets instantiated we
> can pass a URI in, but then a "new" uri needs to be passed in for each
> PUT/POST/GET... The pattern looks something like this
>
> IrisClient client = ...
> ...
> client.put(client.buildUri(...), payload);
>
> I'm not a big fan of making the user first ask the client to build a URI.
> I'm sure there is a better way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bandwidthcom/java-bandwidth-iris/pull/1/files#r34065524>
> .
>



-- 
scottbarstow.com
@scottbarstow